Monsters hunting persistence/src/main/java/cz/fi/muni/pa165/dao/UserDaoImpl.java

```diff
@@ -1,35 +1,70 @@
 package cz.fi.muni.pa165.dao;
 
 import cz.fi.muni.pa165.entity.User;
-import sun.reflect.generics.reflectiveObjects.NotImplementedException;
 
 import javax.inject.Named;
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.PersistenceContext;
+import javax.persistence.NoResultException;
 import java.util.List;
 
+/**
+ * Implementation of UserDao interface
+ * @author Balcirak Peter <peter.balcirak@gmail.com>
+ */
 @Named
 public class UserDaoImpl implements UserDao {
+
+	@PersistenceContext
+	private EntityManager em;
+
 	@Override
-	public void create(User user) {
-		throw new NotImplementedException();
+	public void create(User user) throws IllegalArgumentException {
+		if(user == null){
+			throw new IllegalArgumentException("Parameter User cannot be null!");
+		}
+		em.persist(user);
 	}
 
 	@Override
-	public void delete(User user) {
-		throw new NotImplementedException();
+	public void delete(User user) throws IllegalArgumentException {
```

Comment:
I think it is not necessary to mention that this method throws a runtime exception although it is written in javadoc.